File: go/0283_move_zeros.go
func moveZeroes(nums []int) {
	cur, skipped := 0, 0
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			skipped++
			continue
		}
		nums[cur] = nums[i]
		cur++
	}

	for i := len(nums) - 1; skipped > 0; i-- {
		nums[i] = 0
		skipped--
	}
}

File: python/0283_move_zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        nlen = len(nums)
        if nlen == 0:
            return []

        left = right = 0
        while True:
            while right < nlen and nums[right] == 0:
                right += 1
            if right == nlen:
                break

            nums[left], nums[right] = nums[right], nums[left]
            left += 1
            right += 1

        for i in range(right, nlen):
            nums[i] = 0
